cw_decred/lib/wallet.dart

```diff
@@ -0,0 +1,205 @@
+import 'package:cw_decred/mnemonic.dart';
+import 'package:cw_decred/transaction_history.dart';
+import 'package:cw_decred/wallet_addresses.dart';
+import 'package:cw_decred/unspent.dart';
+import 'package:cw_decred/transaction_priority.dart';
+import 'package:cw_core/crypto_currency.dart';
+import 'package:cw_core/unspent_coins_info.dart';
+import 'package:hive/hive.dart';
+import 'package:mobx/mobx.dart';
+import 'dart:io';
+import 'package:flutter/foundation.dart';
+import 'package:cw_dcrlibwallet/cw_dcrlibwallet.dart';
+import 'package:cw_core/wallet_info.dart';
+import 'package:cw_core/wallet_base.dart';
+import 'package:cw_core/balance.dart';
+import 'package:cw_core/transaction_priority.dart';
+import 'package:cw_core/pending_transaction.dart';
+import 'package:cw_core/transaction_info.dart';
+import 'package:cw_core/pathForWallet.dart';
+import 'package:cw_core/sync_status.dart';
+import 'package:cw_core/node.dart';
+
+part 'wallet.g.dart';
+
+class DecredWallet = DecredWalletBase with _$DecredWallet;
+
+abstract class DecredWalletBase extends WalletBase<Balance, TransactionHistory, TransactionInfo> with Store {
+  DecredWalletBase( SPVWallet spv, WalletInfo walletInfo)
+    : this.spv = spv,
+      balance = ObservableMap<CryptoCurrency, Balance>.of({CryptoCurrency.dcr: spv.balance()}),
+      syncStatus = NotConnectedSyncStatus(),
+    super(walletInfo){
+      transactionHistory = TransactionHistory(spv);
+      walletAddresses = WalletAddressesBase(walletInfo, spv);
+    }
+
+  final SPVWallet spv;
+
+  Future<void> init(WalletInfo walletInfo) async {
+    await transactionHistory.init();
+  }
+
+  @override
+  @observable
+  late ObservableMap<CryptoCurrency, Balance> balance;
+
+  @override
+  @observable
+  SyncStatus syncStatus;
+
+  // @override
+  // set syncStatus(SyncStatus status);
```

Comment:
tbh I have no idea how these observable fields work

Reply:
I don't guess it would be possible to have dcrlibwallet accept an observable? or return something we can observe? Will we need to do long polling to get sync status and I guess balance changes?

Reply:
Yeah, I think will become clearer when we start implementing sync. Can leave as is for now.